sklearn/pipeline.py

```diff
@@ -33,6 +33,13 @@
 __all__ = ["Pipeline", "FeatureUnion", "make_pipeline", "make_union"]
 
 
+def _final_estimator_has(attr):
+    """Check that final_estimator has `attr`.
+
+    Used together with `avaliable_if` in `Pipeline`."""
+    return lambda self: getattr(self._final_estimator, attr) or True
```

Comment:
Would it not be enough to cast to bool (and add a small comment because I had to think quite a bit about it :))

```suggestion
    # we do not use `hasattr` to raise the `AttributeError`
    return lambda self: bool(getattr(self._final_estimator, attr))
```

Reply:
the `or True` is certainly not easy to understand... what's it doing?

Reply:
It's using the same logic as https://github.com/scikit-learn/scikit-learn/blob/beea7838cc8ec61c78255bd9c8b400997e30db32/sklearn/utils/metaestimators.py#L188

If the `or True` is not there, then the truthiness of `self._final_estimator_.attr` would be used to evaluate the check.

I'll leave a comment.

Reply:
I'll update the implementation to  make it less confusing:

```python
if not hasattr(self._final_estimator, attr):
    # raise the original `AttributeError`
    getattr(self._final_estimator, attr)
return True
```